libafl/src/executors/forkserver.rs

```diff
@@ -1635,7 +1635,7 @@ mod tests {
 
         let mut shmem = shmem_provider.new_shmem(MAP_SIZE).unwrap();
         shmem.write_to_env("__AFL_SHM_ID").unwrap();
-        let shmem_buf = shmem.as_slice_mut();
+        let shmem_buf: &mut [u8; MAP_SIZE] = shmem.as_slice_mut().try_into().unwrap();
```

Comment:
Wait, doesn't afl++ put the length of the shmem as the first variable on the map? So in general this is wrong, right?
CC @tokatoka @R9295

Reply:
it's the input length that is put on the first 4 bytes of the shmem. (and it gets updated everytime new inputs are put)
so it has nothing to do with map size

Reply:
So inner workings of Forkserver know about this and it will "just work"?

Reply:
This feels more like a "LengthTaggedMapObserver" to me than a "ConstLengthMapObserver" tbh

Reply:
some maps don't embed the length of the input in the map no? isn't this something that is unrelated to trait that just guarantees the map has a length fixed at compile time?
the way the map is used later on is another problem, right?

Reply:
Yes true, unrelated to this PR, then. But should be fixed / properly typed IMHO

Reply:
Opened issue #2651 for that

Reply:
sounds good thanks